tests/storage/glusterfs/conftest.py

```diff
@@ -65,15 +65,18 @@ def pool_with_glusterfs(host):
 def gluster_disk(host, sr_disk):
     print(">> Format sr_disk %s and mount it" % sr_disk)
     device = '/dev/' + sr_disk
-    host.ssh(['mkfs.ext4', '-F', device])
-    host.ssh(['mkdir', '-p', '/mnt/sr_disk'])
-    host.ssh(['cp', '-f', '/etc/fstab', '/etc/fstab.orig'])
-    host.ssh(['echo', '/dev/sdb /mnt/sr_disk ext4 defaults 0 0', '>>/etc/fstab'])
-    host.ssh(['mount', '/mnt/sr_disk'])
+    hosts = host.pool.hosts
+    for h in hosts:
+        h.ssh(['mkfs.ext4', '-F', device])
+        h.ssh(['mkdir', '-p', '/mnt/sr_disk'])
+        h.ssh(['cp', '-f', '/etc/fstab', '/etc/fstab.orig'])
+        h.ssh(['echo', '/dev/sdb /mnt/sr_disk ext4 defaults 0 0', '>>/etc/fstab'])
+        h.ssh(['mount', '/mnt/sr_disk'])
     yield
-    host.ssh(['cp', '-f', '/etc/fstab.orig', '/etc/fstab'])
-    host.ssh(['umount', '/mnt/sr_disk'])
-    host.ssh(['rmdir', '/mnt/sr_disk'])
+    for h in hosts:
+        h.ssh(['cp', '-f', '/etc/fstab.orig', '/etc/fstab'])
+        h.ssh(['umount', '/mnt/sr_disk'])
+        h.ssh(['rmdir', '/mnt/sr_disk'])
```

Comment:
The sr_disk fixture only guarantees the presence of an available disk on the first host, so this will not fail nicely for other hosts if they don't have such a disk.